### PR TITLE
Swap routing

### PR DESCRIPTION
**Pull request by PawelKrawczyk96**

### CODE REVIEW

File: src/store/consts/utils.ts [hunk 1]
```diff
@@ -557,7 +557,7 @@ export const handleSimulate = async (
   let poolIndex: number = 0
   let isXtoY = false
   let resultWithFee: BN = new BN(0)
-  let result
+  let resault
```
Comment (awojciak) [inline src/store/consts/utils.ts:560]: typo

Comment (PawelKrawczyk96) [reply src/store/consts/utils.ts]: done

File: src/containers/EventsHandlers/market.ts [hunk 2]
```diff
@@ -172,30 +172,17 @@ const MarketEvents = () => {
...
+      for (const pool of pools) {
         marketProgram
-          .getTickmap(new Pair(pools[0].tokenX, pools[0].tokenY, { fee: pools[0].fee.v }))
+          .getTickmap(new Pair(pool.tokenX, pool.tokenY, { fee: pool.fee.v }))
           .then(res => {
-            dispatch(
-              actions.setTickMaps({ index: pools[0].tickmap.toString(), tickMapStructure: res })
-            )
+            dispatch(actions.setTickMaps({ index: pool.tickmap.toString(), tickMapStructure: res }))
           })
```
Comment (awojciak) [inline src/containers/EventsHandlers/market.ts:180]: add catch on the end to get rid of trunk error

File: src/store/consts/utils.ts [hunk 3]
```diff
@@ -563,15 +564,6 @@ export const handleSimulate = async (
       error: errorMessage
     }
   }
-  if (errorMessage.length > 0) {
```
Comment (awojciak) [inline src/store/consts/utils.ts:566]: are you sure this won't break anything, like cases when error message is needed?

Comment (PawelKrawczyk96) [reply src/store/consts/utils.ts:566]: yes i'm sure. With this case we receive error even we can swap tokens.

**Review by awojciak — APPROVED** [review]

**Review by wojciech-cichocki — APPROVED** [review]

**Pull request merged**